account_payment_order/models/account_payment_order.py

```diff
@@ -0,0 +1,616 @@
+                )
+            # Delete existing bank payment lines
+            order.bank_line_ids.unlink()
+            # Create the bank payment lines from the payment lines
+            group_paylines = {}  # key = hashcode
+            for payline in order.payment_line_ids:
+                payline.draft2open_payment_line_check()
+                # Compute requested payment date
+                if order.date_prefered == "due":
+                    requested_date = payline.ml_maturity_date or payline.date or today
+                elif order.date_prefered == "fixed":
+                    requested_date = order.date_scheduled or today
+                else:
+                    requested_date = today
+                # No payment date in the past
+                if requested_date < today:
+                    requested_date = today
+                # inbound: check option no_debit_before_maturity
+                if (
+                    order.payment_type == "inbound"
+                    and order.payment_mode_id.no_debit_before_maturity
+                    and payline.ml_maturity_date
+                    and requested_date < payline.ml_maturity_date
+                ):
+                    raise UserError(
+                        _(
+                            "The payment mode '%(pmode)s' has the option "
+                            "'Disallow Debit Before Maturity Date'. The "
+                            "payment line %(pline)s has a maturity date %(mdate)s "
+                            "which is after the computed payment date %(pdate)s.",
+                            pmode=order.payment_mode_id.name,
+                            pline=payline.name,
+                            mdate=payline.ml_maturity_date,
+                            pdate=requested_date,
+                        )
+                    )
+                # Write requested_date on 'date' field of payment line
+                # norecompute is for avoiding a chained recomputation
+                # payment_line_ids.date
+                # > payment_line_ids.amount_company_currency
+                # > total_company_currency
+                with self.env.norecompute():
+                    payline.date = requested_date
+                # Group options
+                if order.payment_mode_id.group_lines:
+                    hashcode = payline.payment_line_hashcode()
+                else:
+                    # Use line ID as hascode, which actually means no grouping
+                    hashcode = payline.id
+                if hashcode in group_paylines:
+                    group_paylines[hashcode]["paylines"] += payline
+                    group_paylines[hashcode]["total"] += payline.amount_currency
+                else:
+                    group_paylines[hashcode] = {
+                        "paylines": payline,
+                        "total": payline.amount_currency,
+                    }
+            order.recompute()
```

Comment:
```suggestion
            order.env._recompute_all()
```
because it calls a no longer existed method https://github.com/odoo/odoo/blob/16.0/odoo/models.py#L6106